pkg/userdata/userdata.go

```diff
@@ -127,6 +131,42 @@ type Install struct {
 	ExtraDevices []*ExtraDevice `yaml:"extraDevices,omitempty"`
 	Wipe         bool           `yaml:"wipe"`
 	Force        bool           `yaml:"force"`
+	installed    bool
+}
+
+func (i *Install) UnmarshalYAML(unmarshal func(interface{}) error) (err error) {
```

Comment:
I'm a bit torn on this file being the appropriate place to do this bit. Seems like it'd be better placed in block device or something.

Reply:
I am too TBH. My thinking here was that I would only have to do it once here, and the state of the current node could be carried throughout the code via the user data. I can play with the idea of being in block device.

Reply:
plus one for this to be moved to the block device

Reply:
So the issue with it being in the definition of the block device is that `probe.GetDevWithFileSystemLabel` probes all block devices for a filesystem label. I'm not sure I see how we can work that into the block device struct.

Reply:
or some other entity (like block device manager)? but certainly side-effect of unmarshal looks a bit strange, plus what if the userdata is unmarshalled say in osctl for validation or something similar?

Reply:
agree